### PR TITLE
Delete posts

### PR DESCRIPTION
**Pull request by l-palermo**

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -30,6 +30,25 @@ def all
     @posts = Post.all.order("created_at DESC")
   end
 
+  # def destroy
+  #   return @error = true if @post.user_id != session[:user_id]
```
Comment (ajosephides) [inline app/controllers/posts_controller.rb]: Should remove all the commented out bits before merging.

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -30,6 +30,25 @@ def all
...
+  #   @post = Post.find(params[:id])
+  #   Post.where(user_id: session[:user_id]).delete(params[:id])
+  #   redirect_to user_posts_url(@user)
+  # end
+
+  def destroy
+    @user = User.find(session[:user_id])
```
Comment (ajosephides) [inline app/controllers/posts_controller.rb]: I understand your frustration about this section (as discussed) but I am also at a loss of how to flash to the user a message if they are not allowed.
What you have written is how I would have done it (there is routing and so logic sits on the controller).
The only other thing I thought to keep the controller skinny is to have some simple logic on the view that would only give the option to delete the post if it was the user.
Something like this:
`<% if @user.id && @user.id == @post.id %>`
wrapped around the button.

File: spec/features/user_delete_own_posts_spec.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    click_link "New post"
+    fill_in "Message", with: "Hello"
+    click_button "Submit"
+    click_link 'Hello'
+    click_link 'Delete'
+    expect(page).to have_content('Welcome test')
+    expect(page).not_to have_content('Hello')
+  end
+
+  scenario "Can submit posts and view them" do
```
Comment (celfro) [inline spec/features/user_delete_own_posts_spec.rb]: scenario same as test above

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (IndecentDolphin) [reply app/controllers/posts_controller.rb]: Seconded

File: app/controllers/posts_controller.rb (hunk 2 above)
Comment (IndecentDolphin) [reply app/controllers/posts_controller.rb]: Meets the requirement:
see a helpful error message if they try to delete another user's post

**Pull request merged**